src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,195 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO car (model, manufacturers_id) VALUES (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();PreparedStatement
+                preparedStatement = connection.prepareStatement(query,Statement
+                .RETURN_GENERATED_KEYS)) {
+            preparedStatement.setObject(1, car.getModel());
+            preparedStatement.setObject(2, car.getManufacturer().isPresent()
+                    ? car.getManufacturer().get().getId() :
+                    Optional.ofNullable(car.getManufacturer()));
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            while (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            insertDriver(car);
```

Comment:
![image](https://user-images.githubusercontent.com/20326858/224689568-74e7a008-47b1-4b1b-b815-5d69ee312452.png)

Reply:
not fixed